website/docs/getting-started/basics.md

```diff
@@ -8,15 +8,34 @@ odo abstracts Kubernetes concepts into developer focussed terminology. In this d
 
 ### Concepts
 
-#### Component
+* Component
+  A component is a microservice. Each component can be run and deployed separately. Examples of components: Warehouse API Backend, Inventory API, Web Frontend, Payment Backend
+  > A component in odo is an equivalent of a deployment or deploymentconfig in Kubernetes. Creating a new component in odo will create a deployment in Kubernetes or for git and binary based component creation, it will create deploymentconfig.
 * Application
+  An application consists of multiple components which may span over multiple projects, and work individually to build the entire application. Examples of applications: e-Shop, Hotel Reservation System, Online Booking
+  > An application can be considered as an equivalent of labels in Kubernetes that help in grouping a set of resources.
 * Project
+  A project is your source code, tests, and libraries organized in a separate single unit.
+  > A project in odo is an equivalent of a namespace in Kubernetes. Creating a project in odo will create a namespace in Kubernetes with the same name.
 * Context
+  A context is the directory where the source code, tests, and libraries for your component resides. A single context can only contain a single component.
 * URL
+  A URL exposes your component to be accessed outside the cluster.
+  > A URL is an equivalent of a service in Kubernetes.
 * Storage
+  A storage volume is [PVC](https://kubernetes.io/docs/concepts/storage/volumes/#persistentvolumeclaim) which is a way for you to "claim" persistent storage without knowing the details of the environment. Storage volume can persist data across restarts and rebuilds of a component.
+  > A storage volume is an equivalent of PVC in Kubernetes. Creating a storage with odo will create a PVC resource in Kubernetes with the same name.
 * Service
+  A service is another microservice, or a Kubernetes Custom Resource that your component connects to or depends on. Example of services: MariaDB, MySQL
```

Comment:
How would you define an operator-backed service? If I use the term, I will still need to explain what a service is. How do I define a service?
By "microservice" I meant component. Ah, I confused service with link :facepalm:

Reply:
> How would you define an operator-backed service? If I use the term, I will still need to explain what a service is. How do I define a service?

A loose definition of service is a long-running daemon (think a server process), and listening for incoming requests based on which it would do certain actions. But we don't need to explain this in our docs. It's a definition that almost everyone relates to when they read the word "service".

When you write "Operator backed service", add a link to the term Operator. Use https://github.com/operator-framework.